### DISCUSSION TITLE
Using the new App Route Handlers

### DISCUSSION THREAD

**Discussion by c0d3x** [vercel/next.js#51662] [App Router]
How can I use a local JSON file for mocking the get request to a data.json file that I want to use within my component until back-end data is ready?

**Comment by MrAlaminH** [comment]
Here are the steps on how to use a local JSON file for mocking the get request to a data.json file that you want to use within your component until back-end data is ready:

1. Create a local JSON file called `data.json` and add the data that you want to mock. For example:

```json
{
  "data": [
    {
      "id": 1,
      "name": "John Doe"
    },
    {
      "id": 2,
      "name": "Jane Doe"
    }
  ]
}
```

2. In your component, create a variable called `data` and initialize it with the contents of the `data.json` file. For example:

```javascript
const data = require('./data.json');
```

3. Now, you can use the `data` variable as if it was the data from the back-end. For example, you can iterate through the data, or you can use it to populate a list.

4. Once the back-end data is ready, you can update the `data` variable with the new data.

Here is an example of how to use a local JSON file for mocking the get request to a data.json file:

```javascript
const data = require('./data.json');

const AppComponent = {
  constructor() {
    this.data = data;
  },

  render() {
    return (
      <ul>
        {this.data.map(item => (
          <li key={item.id}>{item.name}</li>
        ))}
      </ul>
    );
  }
};
```

This code will first render the data from the `data.json` file. Once the back-end data is ready, you can update the `data` variable with the new data, and the component will render the new data.

**Comment by c0d3x** [comment]
I want to use this as a GET request:
https://nextjs.org/docs/app/building-your-application/routing/router-handlers

I tried doing something like this in combination with the new route-handlers:
https://vercel.com/guides/loading-static-file-nextjs-api-route

But somehow the json data is not parsed properly.

**Reply by c0d3x** [reply]
```TS
import { NextResponse } from 'next/server'
import path from 'path'
import { promises as fs } from 'fs'

export async function GET() {
    // Find the absolute path of the json directory
    const jsonDirectory = path.join(process.cwd(), 'json')

    // Read the json data file data.json
    const fileContents = await fs.readFile(`${jsonDirectory}/data.json`, 'utf8')

    // Parse Data as JSON
    const res = JSON.parse(fileContents)

    // Return the content of the data file in json format
    return NextResponse.json(res)
}
```

So added a JSON.parse before returning the data and now it works.
